vignettes/adrs_pcwg3.Rmd

```diff
@@ -0,0 +1,454 @@
+  
+  # Tumor Marker Response (PCWG3)
+  "PCWG SCHER PROSTATE CANCER 2016", "TMRESP", "TMRESP", "Tumor Marker Response by Investigator", 4, "PCWG3", 4,
+  
+  # Overall Tumor Response (PCWG3)
+  "PCWG SCHER PROSTATE CANCER 2016", "OVRLRESP", "OVRLRESP", "Overall Tumor Response by Investigator CRF", 5, "PCWG3 Collected", 5
+)
+
+```  
+
+### Table : Overall Time Point Response
+
+```{r, eval=TRUE, include=TRUE, message=FALSE}
+
+
+overall_tpr_table <- tribble(
+  ~`Soft Tissue (RECIST 1.1) TPR`, ~`Bone Lesion (PCWG3) TPR`, ~`Overall PCWG TPR`,
+  "PD", "Any", "PD",
+  "Any", "PD", "PD",
+  "NE", "Non-PD, PDu, NED, or NE", "NE",
+  "NED", "Non-PD", "Non-CR/Non-PD",
+  "NED", "PDu", "PDu",
+  "NED", "NED", "NE",
+  "NED", "NE", "NE",
+  "SD", "Non-PD, PDu, NED, or NE", "SD",
+  "Non-CR/Non-PD", "Non-PD, PDu, NED, or NE", "Non-CR/Non-PD",
+  "PR", "Non-PD, PDu, NED, or NE", "PR",
+  "CR", "Non-PD, PDu, or NE", "PR (1)",
+  "CR", "NED", "CR"
+)
+
+# Render Colorful Table with Fixed Footnotes
+overall_tpr_table %>%
+  gt() %>%
+  tab_header(
+    title = "Table 1: Overall Time Point Response",
+    subtitle = "Soft Tissue (RECIST 1.1) TPR, Bone Lesion (PCWG3) TPR, and PCWG Combined TPR"
+  ) %>%
+  cols_label(
+    `Soft Tissue (RECIST 1.1) TPR` = "Soft Tissue (RECIST 1.1)",
+    `Bone Lesion (PCWG3) TPR` = "Bone Lesion (PCWG3)",
+    `Overall PCWG TPR` = "Overall PCWG"
+  ) %>%
+  tab_footnote(
+    footnote = "When no target and non-target lesions are identified at baseline, and no new lesions are identified on-study, the response will be No Evidence of Disease (NED).",
+    locations = cells_column_labels(columns = `Overall PCWG TPR`)  # Correct column name
+  ) %>%
+  tab_footnote(
+    footnote = "Progressive Disease Unconfirmed (PDu): Temporary marker of possible PD where at least 2 new bone lesions are present, but an additional scan is required for confirmation.",
+    locations = cells_column_labels(columns = `Bone Lesion (PCWG3) TPR`)  # Correct column name
+  ) %>%
+  tab_footnote(
+    footnote = "(1) The overall TPR will be PR if target lesions were present at screening.",
+    locations = cells_body(columns = `Overall PCWG TPR`, rows = 11)  # Correct column name
+  ) %>%
+  tab_footnote(
+    footnote = "(2) The overall TPR will be Non-CR/Non-PD if no target lesions were present at screening.",
+    locations = cells_body(columns = `Overall PCWG TPR`, rows = 12)  # Correct column name
+  )
+```
+
+
+
+### Derive Combined Overall Response Records per PCWG3 Guidelines as per above table.
+
+
+```{r, eval=TRUE, include=TRUE, message=FALSE}
+soft_tpr <- adrs %>%
+  filter(RSTESTCD == "SFTSRESP") %>%
+  mutate(AVALC = RSSTRESC
+  )
+  
+``` 
+
+```{r, eval=TRUE, include=TRUE, message=FALSE}
+bone_tpr <- adrs %>%
+  filter(RSTESTCD == "BONERESP") %>%
+  mutate(AVALC = RSSTRESC
+  )
+```  
+
+```{r, eval=TRUE, include=TRUE, message=FALSE}
+overall_tpr <- soft_tpr %>%
```

Comment:
if we derive this overall response , then we need to test this code on some actual data? also is this can be put as a function?